src/library/commissioning_session.cpp

```diff
@@ -181,22 +181,22 @@ void CommissioningSession::HandleJoinFin(const coap::Request &aJoinFin)
     VerifyOrExit((vendorStackVersionTlv = tlvSet[tlv::Type::kVendorStackVersion]) != nullptr, error = Error::kNotFound);
     VerifyOrExit(vendorStackVersionTlv->IsValid(), error = Error::kBadFormat);
 
-    LOG_INFO("received JOIN_FIN.req: vendorName={}, vendorModel={}, vendorSWVversion={}, "
-             "vendorStackVersion={}",
-             vendorNameTlv->GetValueAsString(), vendorModelTlv->GetValueAsString(),
-             vendorSwVersionTlv->GetValueAsString(), utils::Hex(vendorStackVersionTlv->GetValue()));
-
     if (auto provisioningUrlTlv = tlvSet[tlv::Type::kProvisioningURL])
     {
         auto vendorDataTlv = tlvSet[tlv::Type::kVendorData];
         VerifyOrExit(vendorDataTlv != nullptr, error = Error::kNotFound);
         VerifyOrExit(vendorDataTlv->IsValid(), error = Error::kBadFormat);
-        VerifyOrExit(provisioningUrlTlv->GetValueAsString() == mJoinerInfo.mProvisioningUrl, error = Error::kReject);
 
         provisioningUrl = provisioningUrlTlv->GetValueAsString();
         vendorData      = vendorDataTlv->GetValue();
     }
 
+    LOG_INFO("received JOIN_FIN.req: vendorName={}, vendorModel={}, vendorSWVversion={}, "
+             "vendorStackVersion={}, provisioningUrl={}, vendorData={:#04x}",
```

Comment:
vendorData is a `ByteArray` but not an integer, maybe print the hex string?

Reply:
`{:#04x}` will print a hex string

Reply:
`fmt::join()` makes sense. Should we use `{:#02x}` instead? By the way, we have `utils::Hex()` to cast `ByteArray` to a continuous hex string (e.g. `utils::Hex(vendorStackVersionTlv->GetValue())`). Your format looks nice, but it will be inconsistent with the rest. Would mind to use the `utils::Hex()`?

We may define `utils::PrettyHex()` to add delimiters to the hex string when used in logging messages (but not necessary in this PR). Thoughts?

Reply:
Oh, I missed utils::Hex(). I will change it to stay consistent.